src/MySqlConnector/Serialization/MySqlSession.cs

```diff
@@ -548,19 +552,43 @@ private void VerifyConnected()
 			Func<object, X509Certificate, X509Chain, SslPolicyErrors, bool> remoteCertificateCb =
 				(rcbSender, rcbCertificate, rcbChain, rcbPolicyErrors) =>
 				{
-					switch (rcbPolicyErrors)
+					if (cs.SslMode == MySqlSslMode.Preferred || cs.SslMode == MySqlSslMode.Required)
+						return true;
+
+					if ((rcbPolicyErrors & SslPolicyErrors.RemoteCertificateChainErrors) > 0 && cs.CACertificate != null)
 					{
-						case SslPolicyErrors.None:
-							return true;
-						case SslPolicyErrors.RemoteCertificateNameMismatch:
-							return cs.SslMode != MySqlSslMode.VerifyFull;
-						default:
-							return cs.SslMode == MySqlSslMode.Preferred || cs.SslMode == MySqlSslMode.Required;
+						X509Certificate2 caCertificate;
+						try
+						{
+							caCertificate = new X509Certificate2(cs.CACertificate);
+						}
+						catch (CryptographicException ex)
+						{
+							if (!File.Exists(cs.CACertificate))
+								throw new MySqlException("Cannot find CA Certificate File", ex);
+							throw new MySqlException("The CA Certificate File is invalid", ex);
+						}
```

Comment:
I wonder if constructing `caCertificate` should be done outside the callback. I also wonder if any `MySqlException` that is thrown gets passed through, or would it end up being wrapped by, say, a `CryptographicException`? (I haven't tested to see.)